presto-kafka/src/main/java/io/prestosql/plugin/kafka/KafkaInternalFieldDescription.java

```diff
@@ -13,126 +13,191 @@
+         */
+        PARTITION_OFFSET_FIELD("_partition_offset", "Offset for the message within the partition"),
+
+        /**
+         * <tt>_message_corrupt</tt> - True if the row converter could not read the a message. May be null if the row converter does not set a value (e.g. the dummy row converter does not).
+         */
+        MESSAGE_CORRUPT_FIELD("_message_corrupt", "Message data is corrupt"),
+
+        /**
+         * <tt>_message</tt> - Represents the full topic as a text column. Format is UTF-8 which may be wrong for some topics. TODO: make charset configurable.
+         */
+        MESSAGE_FIELD("_message", "Message text"),
+
+        /**
+         * <tt>_message_length</tt> - length in bytes of the message.
+         */
+        MESSAGE_LENGTH_FIELD("_message_length", "Total number of message bytes"),
+
+        /**
+         * <tt>_headers</tt> - The header fields of the Kafka message. Key is a UTF-8 String and values an array of byte[].
+         */
+        HEADERS_FIELD("_headers", "Headers of the message as map."),
+
+        /**
+         * <tt>_key_corrupt</tt> - True if the row converter could not read the a key. May be null if the row converter does not set a value (e.g. the dummy row converter does not).
+         */
+        KEY_CORRUPT_FIELD("_key_corrupt", "Key data is corrupt"),
+
+        /**
+         * <tt>_key</tt> - Represents the key as a text column. Format is UTF-8 which may be wrong for topics. TODO: make charset configurable.
+         */
+        KEY_FIELD("_key", "Key text"),
+
+        /**
+         * <tt>_key_length</tt> - length in bytes of the key.
+         */
+        KEY_LENGTH_FIELD("_key_length", "Total number of key bytes");
+
+        private static final Map<String, InternalFields> BY_COLUMN_NAME =
+                stream(InternalFields.values())
+                        .collect(toImmutableMap(InternalFields::getColumnName, identity()));
+
+        public static InternalFields forColumnName(String columnName)
+        {
+            InternalFields description = BY_COLUMN_NAME.get(columnName);
+            checkArgument(description != null, "Unknown internal column name %s", columnName);
+            return description;
+        }
+
+        private final String columnName;
+        private final String comment;
+
+        InternalFields(
+                String columnName,
+                String comment)
+        {
+            checkArgument(!isNullOrEmpty(columnName), "name is null or is empty");
+            this.columnName = columnName;
+            this.comment = requireNonNull(comment, "comment is null");
+        }
+
+        public String getColumnName()
+        {
+            return columnName;
+        }
     }
 
-    public String getColumnName()
+    /**
+     * Inner class that wraps the {@link InternalFields} enum and add {@link Type}.
+     * With this bean a non static data structure with types from the {@link TypeManager} can be created.
+     */
```

Comment:
I don't know if this is fit for a Javadoc comment.

Reply:
Should I write a non-doc comment instead or just remove it?

Reply:
I would opt for removal.